fastapi/dependencies/utils.py

```diff
@@ -587,13 +587,25 @@ async def solve_dependencies(
         response = Response()
         del response.headers["content-length"]
         response.status_code = None  # type: ignore
-    dependency_cache = dependency_cache or {}
+    if dependency_cache is None:
+        dependency_cache = {}
     sub_dependant: Dependant
     for sub_dependant in dependant.dependencies:
         sub_dependant.call = cast(Callable[..., Any], sub_dependant.call)
         sub_dependant.cache_key = cast(
             Tuple[Callable[..., Any], Tuple[str]], sub_dependant.cache_key
         )
+
+        if sub_dependant.use_cache:
+            # Use a unique object to compare against in case the cached value is None
+            cache_miss = object()
+            cached_value = dependency_cache.get(sub_dependant.cache_key, cache_miss)
+            # If the sub dependant is already cached, skip doing any more work
+            if cached_value is not cache_miss:
+                if sub_dependant.name is not None:
+                    values[sub_dependant.name] = cached_value
+                continue
```

Comment:
If I understand this right, nothing will happen in the end if `sub_dependant.name is None`, right? So maybe when that is the case we shouldn't even check the cache? I'm thinking we could change the `if` on L599 to include the `if` from L605, if you see what I mean:

```suggestion
        if sub_dependant.use_cache and sub_dependant.name is not None:
            # Use a unique object to compare against in case the cached value is None
            cache_miss = object()
            cached_value = dependency_cache.get(sub_dependant.cache_key, cache_miss)
            # If the sub dependant is already cached, skip doing any more work
            if cached_value is not cache_miss:
                values[sub_dependant.name] = cached_value
                continue
```

Reply:
Hm, not exactly _nothing_, since it will still hit that `continue` and skip doing any work here the value was cached. I'm not sure if this is a real thing that can happen though. Do you know of any cases where there would not be a `sub_dependant.name`?

Reply:
So from reading a bit of the code it seems like dependencies that are added to an api route directly won't have a name. But I think we should still skip getting them again if they are in the cache.

https://fastapi.tiangolo.com/tutorial/dependencies/dependencies-in-path-operation-decorators/#add-dependencies-to-the-path-operation-decorator

Reply:
@xkabylgSICKAG right, we should leave this block of code as it is. We don't store the value. but we still need to skip solving this dependency.